openapps/opencoap/opencoap.c

```diff
@@ -846,19 +873,23 @@ owerror_t opencoap_options_encode(
     return E_SUCCESS;
 }
 
-coap_option_iht* opencoap_find_option(coap_option_iht* array, uint8_t arrayLen, coap_option_t option) {
-    uint8_t i;
-
-    if (array == NULL || arrayLen == 0) {
-        return NULL;
-    }
+uint8_t opencoap_find_option(coap_option_iht* array, uint8_t arrayLen, coap_option_t option, uint8_t* startIndex) {
+   uint8_t i, j;
 
-    for (i = 0; i < arrayLen; i++) {
-        if (array[i].type == option) {
-            return &array[i];
-        }
-    }
-    return NULL;
+   for (i=0; i<MAX_COAP_OPTIONS; i++) {
+      if (array[i].type == option) {
+         // skip until different type
+         for (j=i; j<MAX_COAP_OPTIONS && array[j].type == option; j++) {}
```

Comment:
I think it would be easier to read the code if you had a single loop and a counter incremented each time a condition is true. Would that not work?